airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_bank_accounts.py

```diff
@@ -0,0 +1,563 @@
+    return json.loads(response_builder.build().body)
+
+
+def _read(
+    config_builder: ConfigBuilder,
+    sync_mode: SyncMode,
+    state: Optional[Dict[str, Any]] = None,
+    expecting_exception: bool = False
+) -> EntrypointOutput:
+    catalog = _catalog(sync_mode)
+    config = config_builder.build()
+    return read(_source(catalog, config), config, catalog, state, expecting_exception)
+
+
+def _assert_not_available(output: EntrypointOutput) -> None:
+    # right now, no stream statuses means stream unavailable
+    assert len(output.get_stream_statuses(_STREAM_NAME)) == 0
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class FullRefreshTest(TestCase):
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_record(self, http_mocker: HttpMocker) -> None:
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _customers_request().with_expands(_EXPANDS).with_created_gte(_A_START_DATE).with_created_lte(_NOW).with_limit(100).build(),
+            _customers_response()
+            .with_record(
+                _a_customer()
+                .with_field(
+                    _SOURCES_FIELD,
+                    _as_dict(
+                        _bank_accounts_response()
+                        .with_record(_a_bank_account())
+                        .with_record(_a_bank_account())
+                    )
+                )
+            )
+            .with_record(
+                _a_customer()
+                .with_field(_SOURCES_FIELD, _as_dict(_bank_accounts_response().with_record(_a_bank_account())))
+            ).build(),
+        )
+
+        output = self._read(_config().with_start_date(_A_START_DATE))
+
+        assert len(output.records) == 3
+
+    @HttpMocker()
+    def test_given_source_is_not_bank_account_when_read_then_filter_record(self, http_mocker: HttpMocker) -> None:
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _customers_request().with_expands(_EXPANDS).with_created_gte(_A_START_DATE).with_created_lte(_NOW).with_limit(100).build(),
+            _customers_response()
+            .with_record(
+                _a_customer()
+                .with_field(
+                    _SOURCES_FIELD,
+                    _as_dict(
+                        _bank_accounts_response()
+                        .with_record(_NOT_A_BANK_ACCOUNT)
+                    )
+                )
+            ).build(),
+        )
+
+        output = self._read(_config().with_start_date(_A_START_DATE))
+
+        assert len(output.records) == 0
+
+    @HttpMocker()
+    def test_given_multiple_bank_accounts_pages_when_read_then_query_pagination_on_child(self, http_mocker: HttpMocker) -> None:
+        _given_events_availability_check(http_mocker)
+        http_mocker.get(
+            _customers_request().with_expands(_EXPANDS).with_expands(_EXPANDS).with_created_gte(_A_START_DATE).with_created_lte(_NOW).with_limit(100).build(),
```

Comment:
Did you mean to call `.with_expands(_EXPANDS)` twice?

Reply:
Nop as the second calls overwrite the first one. Thanks for pointing that out!